### PR TITLE
5.10 hdmi cec rework

### PR DESCRIPTION
**Pull request by popcornmix**
This switches our CEC support to the version @mripard has cleaned up for upstreaming.
In testing it worked for me using kms on Pi3 and Pi4 in a range of HDMI modes.

### CODE REVIEW

**Comment by popcornmix** [comment]
@pelwell I've attempted to disable by default and enable in the overlay but it's not working.
I think target should reference `interrupt-controller` but that's not working.

File: arch/arm64/configs/bcmrpi3_defconfig [hunk 1]
```diff
@@ -1216,6 +1216,7 @@ CONFIG_MAXIM_THERMOCOUPLE=m
 CONFIG_MAX31856=m
 CONFIG_PWM_BCM2835=m
 CONFIG_PWM_PCA9685=m
+CONFIG_BRCMSTB_L2_IRQ=y
```
Comment (6by9) [inline arch/arm64/configs/bcmrpi3_defconfig]: Why enable it for Pi3 defconfig? That seems odd?

Comment (popcornmix) [reply arch/arm64/configs/bcmrpi3_defconfig]: Good point - I'll remove it.

Comment (popcornmix) [reply arch/arm64/configs/bcmrpi3_defconfig]: (not entirely sure of the use of bcmrpi_3_defconfig - we use bcm2711_defconfig for official 64-bits on Pi3 and Pi4).

Comment (pelwell) [reply arch/arm64/configs/bcmrpi3_defconfig]: We could get rid of it, except that people are using it and might want the slightly smaller kernel on Pi 3.

File: arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts [hunk 2]
```diff
@@ -115,6 +115,20 @@
 		};
 	};
 
+	fragment@15 {
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts]: This is the wrong overlay - you want vc4-kms-v3d-pi4.

Comment (popcornmix) [reply arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts]: Ah...

**Comment by pelwell** [comment]
Does the overlay now successfully enable the interrupt controllers?

**Review by mripard — COMMENTED** [review]
We should probably not merge 961cd2f98034, or introduce it back.

It was meant to ease the rebasing of my branch, but it still looks useful

**Comment by popcornmix** [comment]
Added "drm/vc4: Reading the hotplug register is only valid if no GPIO defined" back.

Yes, overlay changes appear successful.
We mask CEC interrupt on firmware and interrupts work on arm when vc4 overlay is enabled.
We do not mask CEC interrupt on firmware when vc4 overlay is not enabled and firmware cec works.

**Comment by popcornmix** [comment]
Updated to v2 of patches. cec now works on second hdmi port. Removed WIP.

**Comment by popcornmix** [comment]
[edit] ignore that - I'm on an older firmware that doesn't disable its handling of cec interrupts

**Comment by popcornmix** [comment]
I think this is good. I believe it's been accepted upstream and testing on Pi3 and Pi4 with kms worked.

**Comment by popcornmix** [comment]
@pelwell okay to merge?

**Comment by 6by9** [comment]
I rebased my tree this morning (still 5.10.13) and I appear to not be able to enable i2c_vc or i2c_arm successfully
```
[    6.153644] i2c-bcm2835 fe804000.i2c: Could not request IRQ
```
This is the main DT change going, but I can't immediately see any issue with it.

I'm doing a quick roll-back to try and confirm. Whilst that's building I'll do an rpi-update to check 5.10.14 out.

**Comment by 6by9** [comment]
Sorry, I should add that it was when vc4-kms-v3d was loaded as well.

**Comment by 6by9** [comment]
5.10.14-v7l from rpi-update files also fail with ```Could not request IRQ```.
I get /dev/i2c-11 and /dev/i2c-12 for the HDMI devices, but not /dev/i2c-1.
Remove ```dtoverlay=vc4-kms-v3d``` from config.txt and /dev/i2c-1 returns.

**Comment by 6by9** [comment]
98a26692449ad74be15a25c05bb5be5396aaea4f (with a couple of WIP patches on top) is good.

**Pull request merged**